pub-critic/src/app/utils/UserContext.tsx

```diff
@@ -0,0 +1,200 @@
+
+  const getMyData = async () => {
+    const response = await getMe();
+    if (response) {
+      setUser(response);
+      return;
+    }
+    setUser(undefined);
+  };
+
+  const updateLikes = (id: number, type: number) => {
+    const possibleDislike = dislikes?.find((dislike) => dislike === id);
+    if (possibleDislike) {
+      updateDislikes(id, 0);
+    }
+
+    if (!likes) return;
+    if (type === 1) {
+      setLikes((prev) => [...prev, id]);
+      return;
+    }
+    setLikes((prev) => prev.filter((like) => like !== id));
+  };
+
+  const updateFavourites = (favourite: Favourite) => {
+    if (!favourites) return;
+    const possibleFavourite = favourites.find(
+      (fav) => fav.gameId === favourite.gameId
+    );
+    if (possibleFavourite) {
+      setFavourites((prev) =>
+        prev.filter((fav) => fav.gameId !== favourite.gameId)
+      );
+      return;
+    }
+    setFavourites((prev) => [...prev, favourite]);
+  };
+
+  const updateDislikes = (id: number, type: number) => {
+    const possibeLike = likes?.find((like) => like === id);
+    if (possibeLike) {
+      updateLikes(id, 0);
+    }
+
+    if (!dislikes) return;
+    if (type === 1) {
+      setDislikes((prev) => [...prev, id]);
+      return;
+    }
+    setDislikes((prev) => prev.filter((dislike) => dislike !== id));
+  };
+
+  const logout = () => {
+    localStorage.removeItem("jwtToken");
+    localStorage.removeItem("time");
+    setUser(undefined);
+  };
+
+  const getMyFavouritesData = async () => {
+    const response = await getMyFavourites();
+    if (response) {
+      setFavourites(response);
+      return;
+    }
+    setFavourites([]);
+  };
+
+  const getMyLikesData = async () => {
+    const response = await getLikesAndDislikes();
+    if (response) {
+      setLikes(response.likes);
+      setDislikes(response.dislikes);
+      return;
+    }
+    setLikes([]);
+    setDislikes([]);
+  };
+
+  const initLoad = async () => {
+    console.log("init");
+    const user = localStorage.getItem("jwtToken");
+    const loggInDate = localStorage.getItem("time");
+    if (!user || !loggInDate) {
+      setLoading(false);
+      console.log("no user");
+      return;
+    }
+    const time = new Date(loggInDate).getTime();
+    console.log(time, user);
+    if (Date.now() - time > 1000 * 3600 * 24) {
+      logout();
+      console.log("logout");
+      setLoading(false);
+      return;
+    }
+    await getMyData();
+  };
+
+  useEffect(() => {
+    initLoad();
+  }, []);
+
+  useEffect(() => {
+    console.log(loading);
+  }, [loading]);
+
+  useEffect(() => {
+    onUserLoad();
+  }, [user]);
+
+  const onUserLoad = async () => {
+    setLoading(true);
+    if (!user) {
+      if (!localStorage.getItem("jwtToken")) {
+        setLoading(false);
+        return;
+      }
+      return;
+    }
+    await getMyFavouritesData();
+    await getMyLikesData();
+    setLoading(false);
+  };
+
+  return !loading ? (
```

Comment:
```
if (!loading) {
return  <UserContext.Provider...
}

return (
<div>...
)
```

Reply:
this will be changed in future pr-s to something else so client call doesnt block whole site